bluer/src/mesh/types.rs

```diff
@@ -0,0 +1,98 @@
+pub use btmesh_common::{
+    address::{Address, UnicastAddress},
+    InsufficientBuffer, ModelIdentifier,
+};
+use btmesh_common::{opcode::Opcode, ParseError};
+use btmesh_models::{Message as ConcreteMessage, Model as ConcreteModel};
+use core::fmt::Debug;
+
+/// Bluetooth Mesh Message
+pub trait Message {
+    /// Returns opcode of the message
+    fn opcode(&self) -> Opcode;
+    /// Emit message parameters
+    fn emit_parameters(&self, xmit: &mut Vec<u8>);
+}
+
+/// Bluetooth Mesh Model
+pub trait Model: Sync + Send + Debug {
+    /// Returns model identifier
+    fn identifier(&self) -> ModelIdentifier;
+    /// Returns whether model supports subscription
+    fn supports_subscription(&self) -> bool;
+    /// Returns whether model supports publication
+    fn supports_publication(&self) -> bool;
+
+    /// Parses message opcode and parameters
+    fn parse<'m>(opcode: Opcode, parameters: &'m [u8]) -> Result<Option<Box<dyn Message + 'm>>, ParseError>
```

Comment:
Why without &self access? Might be right, but I don't understand, please explain.